pkgs/by-name/ad/ada-url/conditional-dl.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/cmake/CPM.cmake b/cmake/CPM.cmake
+index ad6b74a8..079eed46 100644
+--- a/cmake/CPM.cmake
++++ b/cmake/CPM.cmake
+@@ -16,9 +16,11 @@ endif()
+ # Expand relative path. This is important if the provided path contains a tilde (~)
+ get_filename_component(CPM_DOWNLOAD_LOCATION ${CPM_DOWNLOAD_LOCATION} ABSOLUTE)
+ 
+-file(DOWNLOAD
+-     https://github.com/cpm-cmake/CPM.cmake/releases/download/v${CPM_DOWNLOAD_VERSION}/CPM.cmake
+-     ${CPM_DOWNLOAD_LOCATION} EXPECTED_HASH SHA256=${CPM_HASH_SUM}
+-)
++if(NOT (EXISTS ${CPM_DOWNLOAD_LOCATION}))
```

Comment:
@lemire should we fix this on Ada?

Reply:
CPM has regular releases and  [CPM.cmake](https://github.com/cpm-cmake/CPM.cmake/releases/tag/v0.39.0) has a recent update. I don't know if the latest release would solve the problem that this patch solves. If not, the proper action would be to submit a patch to CPM.

Modifying CPM.cmake ourselves is not a good idea. We should rely on releases.

Reply:
This is not the "main" CPM.cmake file, but a variation on their provided `get_cpm.cmake` script. However, I have still filed https://github.com/cpm-cmake/CPM.cmake/issues/554 and if they approve such a change I can also send a patch to ada-url to update their cmake file.

Reply:
That would be fantastic!